src/messaging/ExchangeContext.cpp

```diff
@@ -218,7 +218,7 @@ CHIP_ERROR ExchangeContext::SendMessage(Protocols::Id protocolId, uint8_t msgTyp
         }
 #endif
 
-        if (err != CHIP_NO_ERROR && IsResponseExpected())
+        if (err != CHIP_NO_ERROR && sendFlags.Has(SendMessageFlags::kExpectResponse) && !IsGroupExchangeContext())
```

Comment:
Again: this "expect a response and not a group" should be factored out into a single boolean above, instead of being copy/pasted.

Reply:
Added a boolean to record whether the message starts the response timer.